### PR TITLE
Clean up code

### PR DESCRIPTION
**Pull request by Innarticles**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,33 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  # before_action :authenticate_user!
```
Comment (Innarticles) [inline app/controllers/application_controller.rb]: Remove commented code

Comment (Innarticles) [reply app/controllers/application_controller.rb]: same

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,4 +1,33 @@
...
+  # @city = request.location.city
+  before_action :set_variables
+  before_action :set_category
+  before_action :prepare_meta_tags
+  layout :layout_by_resource
+
+  
+  end
```
Comment (Innarticles) [inline app/controllers/application_controller.rb]: Unexpected end of Class. I believe the code below all belongs to this class.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,4 +1,33 @@
...
 
 class ApplicationController < ActionController::Base
+  # before_action :authenticate_user!
+  # @city = request.location.city
+  before_action :set_variables
+  before_action :set_category
+  before_action :prepare_meta_tags
+  layout :layout_by_resource
+
+
```
Comment (Innarticles) [inline app/controllers/application_controller.rb]: Remove extra lines. Only one line should separate code blocks.

File: app/controllers/home_controller.rb [hunk 4]
```diff
@@ -2,5 +2,7 @@
 
 # Renders the home page.
 class HomeController < ApplicationController
-  def index; end
+  layout 'landing_page'
```
Comment (Innarticles) [inline app/controllers/home_controller.rb]: add a space between line 5 and 6.

You can also do 
`def ; end`

File: app/controllers/notes_controller.rb [hunk 5]
```diff
@@ -1,59 +1,162 @@
 # frozen_string_literal: true
 
 class NotesController < ApplicationController
+  layout 'react'
+  #before_action :get_category, except: %i[test trash state search tags]
```
Comment (Innarticles) [inline app/controllers/notes_controller.rb]: Remove commented code

File: app/models/note.rb [hunk 6]
```diff
@@ -1,4 +1,78 @@
...
+  validates :content, length: { minimum: 200 }
+  # sort the note to most recent
+  scope :recent, -> { order(created_at: :desc) }
+  scope :most_recent, -> { order(title: DESC) }
+  self.per_page = 5
+  
+  def should_generate_new_friendly_id?
+    slug.blank? || title_changed?
+  end
+  def slug=(value)
```
Comment (Innarticles) [inline app/models/note.rb]: Separate methods with one empty line.

File: app/models/note.rb [hunk 7]
```diff
@@ -1,4 +1,78 @@
...
+  end
+
+  after_create do
+    note = Note.find_by(id: id)
+    hashtags = content.scan(/#\w+/)
+    hashtags.uniq.map do |hashtag|
+      tag = Tag.find_or_create_by(keyword: hashtag.downcase.delete('#'))
+      note.tags << tag
+    end
+  end
```
Comment (Innarticles) [inline app/models/note.rb]: Separate blocks with one empty line.

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -1,11 +1,10 @@
 <%# The "application" layout is the default layout. It extends "base"  %>
```
Comment (Innarticles) [inline app/views/layouts/application.html.erb]: Remove

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -1,11 +1,10 @@
...
 <%# to include things like header, footer, and alerts that are common  %>
 <%# to most (but not all) views.                                       %>
+<%= javascript_include_tag "application" %>
 
 <%= render("shared/navbar") %>
-<p class="notice"><%= notice %></p>
-<p class="alert"><%= alert %></p>
-
+<%# REMOVE USER STATUS CHECKER %>
```
Comment (Innarticles) [inline app/views/layouts/application.html.erb]: same

File: app/views/notes/tags.html.erb [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    <li class="nav-item">
+    </li>
+</div>
+     
+<tbody>
+    <% @notes.each do |note| %>
+    <tr>
+        <td><%= note.title %></td>
+        <td><%= note.tags  %></td>
+
```
Comment (Innarticles) [inline app/views/notes/tags.html.erb]: extra space

File: app/views/notes/test.html.erb [hunk 11]
```diff
@@ -0,0 +1 @@
+<input id="test">
```
Comment (Innarticles) [inline app/views/notes/test.html.erb]: Useful file?

File: app/views/tags/index.html.erb [hunk 12]
```diff
@@ -0,0 +1,73 @@
+<% provide(:title, "Listing tags") %>
+
+
+
+<section class="content">
```
Comment (Innarticles) [inline app/views/tags/index.html.erb]: too much space

File: config/environment.rb [hunk 13]
```diff
@@ -3,5 +3,14 @@
 # Load the Rails application.
 require_relative 'application'
 
+ActionMailer::Base.smtp_settings = {
```
Comment (Innarticles) [inline config/environment.rb]: indent

File: config/environments/production.rb [hunk 14]
```diff
@@ -76,20 +76,21 @@
...
-  }
-  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.delivery_method = :smtp
   config.action_mailer.default_url_options = {
-    host: 'blacknote',
-    protocol: 'https'
+    host: "http://TODO_PUT_YOUR_DOMAIN_HERE"
   }
+  config.action_mailer.raise_delivery_errors = true
+  # config.action_mailer.default_url_options = {
```
Comment (Innarticles) [inline config/environments/production.rb]: Remove

File: app/views/notes/test.html.erb (hunk 11 above)
Comment (ifebrand6) [reply app/views/notes/test.html.erb]: no, I use to test my code before implementation

**Pull request merged**